File: Iterator/Golang/UserCollection.go
package main

type UserCollection struct {
	Collection []*User
}

func (userCollection *UserCollection) createIterator() IIterator {
	return &UserIterator{
		Position: 0,
		Collection: userCollection.Collection,
	}
}

func (userCollection *UserCollection) add(user *User)  {
	userCollection.Collection = append(userCollection.Collection, user)
}

File: Iterator/PHP/UserCollection.php
<?php

require_once "ICollection.php";
require_once "User.php";
require_once "UserIterator.php";

class UserCollection implements ICollection {
    private array $collection;

    public function createIterator() {
        return new UserIterator($this->collection);
    }

    public function add(User $user) {
        $this->collection[] = $user;
    }
}